tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -420,23 +421,48 @@ def read_model(path_to_xml):
     mean_data = deepcopy(graph.graph['mf']) if 'mf' in graph.graph else None
     input_names = deepcopy(graph.graph['input_names']) if 'input_names' in graph.graph else []
 
-    prepare_emit_ir(graph=graph,
-                    data_type=graph.graph['cmd_params'].data_type,
-                    output_dir=argv.output_dir,
-                    output_model_name=argv.model_name,
-                    mean_data=mean_data,
-                    input_names=input_names,
-                    meta_info=non_default_params,
-                    use_temporary_path=True)
-
-    # This graph cleanup is required to avoid double memory consumption
-    graph.clear()
-
     output_dir = argv.output_dir if argv.output_dir != '.' else os.getcwd()
     orig_model_name = os.path.normpath(os.path.join(output_dir, argv.model_name))
 
-    fem = FrontEndManager()
-    func = read_model(orig_model_name + "_tmp.xml")
+    def clear_tmp_ir_files():
+        for suf in [".xml", ".bin", ".mapping"]:
+            # remove existing files
+            path_to_file = orig_model_name + "_tmp" + suf
+            if os.path.exists(path_to_file):
+                os.remove(path_to_file)
+
+    try:
+        prepare_emit_ir(graph=graph,
+                        data_type=graph.graph['cmd_params'].data_type,
+                        output_dir=argv.output_dir,
+                        output_model_name=argv.model_name,
+                        mean_data=mean_data,
+                        input_names=input_names,
+                        meta_info=non_default_params,
+                        use_temporary_path=True)
+    except Error as err:
+        clear_tmp_ir_files()
+        raise Error('Error occurred during serialization of temporary IR: {}'.format(
+            str(err),
+        )) from err
+    except Exception as err:
+        clear_tmp_ir_files()
+        raise Exception('Exception occurred during serialization of temporary IR: {}'.format(
+            str(err),
+        )) from err
+    finally:
+        # This graph cleanup is required to avoid double memory consumption
+        graph.clear()
+
+    try:
+        fem = FrontEndManager()
+        func = read_model(fem, orig_model_name + "_tmp.xml")
+    except Exception as err:
+        raise Exception('Exception occurred while reading of temporary IR: {}'.format(
+            str(err),
+        )) from err
+    finally:
+        clear_tmp_ir_files()
```

Comment:
Now all of these messages about failure in serialization and deserialization of the temporal IR say nothing to our user. Also, it is useless to ours because we can localize place of the error by the call stack.
I suggect to handle all exceptions by just the single `except Exception as err:` and print error as is. Please simplify - I think overcomplication of this logic not worth

Reply:
Agreed. Corrected, left only one `except` block.

The reason why in the first place i added separate `except Error...` is that long time ago i was told to prefer internal `Error` which is gracefully handlesd by mo and does not print backtrace into log unless user explicitly specified `--log_level=DEBUG` and we have separate `except` for `Error` and `Exception` in some other places: https://github.com/openvinotoolkit/openvino/blob/558714ccdd04fba432d761103051dd4aa522dc1b/tools/mo/openvino/tools/mo/utils/class_registration.py#L301-L314

So I thought if somewhere inside our custom `Error` was raised handle and propagate it as is without mixing with python base Exception. But, indeed, i agree with you, this does not give any useful information for users and for us. Left a singe `except` block.